lib/operations/find_and_modify.js

```diff
@@ -22,17 +22,16 @@ class FindOneAndDeleteOperation extends OperationBase {
   execute(callback) {
     const coll = this.collection;
     const query = this.query;
-    let sort = this.sort;
+    const sort = formattedOrderClause(this.sort);
     const doc = this.doc;
-    const options = this.options;
+    const options = Object.assign({}, this.options);
 
     // Create findAndModify command object
     const queryObject = {
       findAndModify: coll.s.name,
```

Comment:
we have public accessors for collectionName, should we use that instead of this private state variable?